### DISCUSSION TITLE
igx-grid - Add row attributes

### DISCUSSION THREAD

**Discussion by mikerentmeister** [IgniteUI/igniteui-angular#11334] [Q&A] [answered]
Does anyone know if there is a way to add an attribute to an `igx-grid-row` when it's rendered? I know that a class can be added via `rowClasses`, but I would like to add an attribute. Maybe some type of event that's emitted when a row is rendered, so that you could set an attribute with `Renderer2`?

My main use case is that I want to add an attribute called `row-key` to the row record, which would be the `primaryKey` value. For example, a row with an pk of 15 would be able to render `<igx-grid-row row-key="15" ... />`. I cannot use `rowClasses`, because the class names being applied must be static.

**Comment by ChronosSF** [comment]
There is no such built-in way currently, however, you could achieve your scenario by getting the data index (attribute data-rowindex ) and taking the primary key from the data record at that index in the grid data view.

**Comment by mikerentmeister** [comment, marked as answer]
If anyone comes here in the future and wants to do something like this, this is how I achieved it:

```
this.grid.verticalScrollContainer.chunkLoad.pipe(takeUntil(this._destroySubject)).subscribe(() => {
  const rowList = this.grid.dataRowList;
  rowList.forEach(row => {
    if ((row as any).automationKey !== row.key) {
      (row as any).automationKey = row.key;
      this._renderer2.setAttribute(row.element.nativeElement, 'data-ex-key', row.key);
      const entityName = row.data.entityId || row.data.name;
      if (entityName) {
        this._renderer2.setAttribute(row.element.nativeElement, 'data-ex-name', entityName);
      }
    }
  });
});
```
